### ISSUE TITLE
Missing shipping instructions on PDF-invoices

### ISSUE THREAD

**Issue by filipefurtad0** [openfoodfoundation/openfoodnetwork#5789]
## Description

Shipping instructions appear on email-generated order confirmations, but are absent from the PDF version of the respective invoices.

## Expected Behavior

All PDF-invoices should contain shipping information, just like the email-embedded order confirmations.

## Actual Behaviour

Only email-embedded order confirmations have this information.

## Steps to Reproduce

1. Admin: Create a shipping method with shipping instructions. Open an order cycle with at least one item with available stock.
2. Customer: Place an order on that order cycle.
3. Admin: Generate PDF-invoices by:
 a) - Bulk printing an invoice (pdf), on the Orders page (/admin/orders?q[s]=completed_at+desc)
 On order edit page, (/admin/orders/Rxxxxxxx/edit)
 b) Print invoice (pdf)
 c) Send the invoice (pdf per email)
 d) Resend (email, no pdf)
4. Verify that a) b) c) do not show shipping instruction -> the bug.
5) Verify d) does.

## Animated Gif/Screenshot

- Order confirmation per email:

![image](https://user-images.githubusercontent.com/49817236/87885624-3920ea00-ca0f-11ea-8eef-350d0c2b09ff.png)

- PDF-invoice:

![image](https://user-images.githubusercontent.com/49817236/87885709-65d50180-ca0f-11ea-9a50-c1d7be2d518a.png)

## Workaround

Consider the info displayed on checkout and/or see info received per Email. 

## Severity

bug-s4: it's annoying, but you can use it

## Your Environment

* Version used: v3.1.0
* Browser name and version: Firefox 78
* Operating System and version (desktop or mobile): Desktop/Ubuntu

## Possible Fix

Observed while testing PR #5712

**Comment by luisramos0** [comment]
I think this needs to be changed to enhancement (papercut) and verified by @openfoodfoundation/train-drivers-product-owners if it's really something desired.

**Comment by RachL** [comment]
Yes, I'm not sure it is. But we need to ask the users to understand it. Definitely not a bug but can keep the template, I think everything is well described :)

**Comment by filipefurtad0** [comment]
Yes, this was agreed to be a feature request, and not a bug [here](https://github.com/openfoodfoundation/openfoodnetwork/issues/5712#issuecomment-662018785). I did not mean to suggest this as an improvement either, I just misinterpreted my findings while testing.

What would be the best approach here @RachL ? Perhaps closing the issue for now and, in case there is need from the community, than re-open using the feature template?

**Comment by RachL** [comment]
@filipefurtad0 yeah user feedback would be needed here. Maybe other people have more strong opinion about it. Can you share the idea in #product-curation and if there is good feedback we can add it to the list of potential papercuts.

**Comment by tschumilas** [comment]
I notice that different information appears on the 2 different invoice templates  (selected at configuration).  For example -  order cycle name appears on the one we use in OFN-CAN now.  This is critical for our users right now because it tells consumers the farmers' market they purchased at.  (Many producers sell at multiple markets,  and so they create different stores (order cycles) for each market).    A week ago  I switched the template to the other one -  because I think its more nicely layed out,  has the enterprise logo...   but I immediately had users ask for it to be switched back because they need order cycle name on it.    So.... just echoing,   I think this is a more complicated thing than what is immediately apparant.   Maybe a wishlist item @filipefurtad0  and then we can all wade in on what info should be on the invoice -   and then make sure both templates have the same info?  (Gosh it would be so amazing if users had the potential to control whats on their invoice -  I know,  a different issue.)

**Comment by filipefurtad0** [comment]
Hey @tschumilas ,

Sure, I see no further comments on [Discourse](https://community.openfoodnetwork.org/t/add-customizable-information-to-pdf-invoices/2005) so I've updated the subject/description to reflect your suggestion and make it more clear for the Dev who pics up this issue.

Thanks again for your feedback :+1:

**Comment by tschumilas** [comment]
This is fantastic  - thanks so much @filipefurtad0

**Comment by RachL** [comment]
@tschumilas @filipefurtad0 in France we are telling our users that OC names are only for internal purposes, and that no customers will see them. So I'm wondering if we can add them to the invoice. Some hubs are not giving a proper name to their OC's.

@lin-d-hop @kirstenalarsen what do you think?

**Comment by tschumilas** [comment]
I don't mean the OC name - sorry -   quite right.  We also tell users this is for their use only.   I mean the 'delivery details'  or the 'ready for'  field.   A user with multiple pickup spots,  for example,  will often created different OCs for different pickup locations (or different farmers markets).  They like having the delivery detail phrase (ie the 'ready for' phrase)  appear on the invoice.  So -  the customer facing phrase,  not the user faced order cycle 'name'.  So sorry -  I shouldn't have used the wrong term.  I mean the phrase in the screenshot below.  I think it,  along with delivery details,  should appear on both invoice templates. 
![screenshot-openfoodnetwork ca-2020 08 25-17_43_40](https://user-images.githubusercontent.com/13171778/91231138-c46e5900-e6fa-11ea-96c3-d6d9a1e87758.png)

**Comment by tschumilas** [comment]
just being really clear now (so sorry) -   the 'alternate invoice' option (in configuration) is not appropriate for North America - where tax is NOT included in price.  So in Canada (and I assume the US),  we have not enabled this option.   So we just use the default invoice option.  The first PDF below is the default invoice option.  See that I have circled how the content from the 'delivery details' field above shows on our invoices now.  The second PDF is what our invoices look like now if we select the 'alternate invoice' type.  No delivery details.    I'm just saying that there is no reason (that I see) for the two PDF invoices to be different,  (except for the tax calculation).     Whatever we add to one,  we should add to the other. 
[invoice-R814570167-when 'use alternative format' is NOT selected.pdf](https://github.com/openfoodfoundation/openfoodnetwork/files/5126621/invoice-R814570167-when.use.alternative.format.is.NOT.selected.pdf)
[invoice-R814570167.pdf (1) - when 'use alternatate formal' IS selected.pdf](https://github.com/openfoodfoundation/openfoodnetwork/files/5126623/invoice-R814570167.pdf.1.-.when.use.alternatate.formal.IS.selected.pdf)

**Comment by filipefurtad0** [comment]
Thanks for the feedback @RachL and @tschumilas for clarifying.

I'm afraid I was also not quite aware of any implications the OC name may have on the invoices. I was a bit unsure about what was displayed where so I ran a small test (on version v.3.2.4). I've set the different **fields in the backoffice** and added a hashtag, so we know where they come from, as described below:


- named the OC in order cycles page - "OC name #/order_cycles"
![image](https://user-images.githubusercontent.com/49817236/91308672-89762f00-e7a7-11ea-9774-32549d75672f.png)

- added info on the Outgoing section on the two sections:

![image](https://user-images.githubusercontent.com/49817236/91309245-41a3d780-e7a8-11ea-9e8a-3bc7e5c3d8ee.png)

replacing "ready for" and the "pick-up instructions" for content with the "#outgoing" label, as below:

![image](https://user-images.githubusercontent.com/49817236/91309171-2f299e00-e7a8-11ea-8582-f6fd627241a8.png)

- added a description on the shipping method - "#shipping method home delivery"
![image](https://user-images.githubusercontent.com/49817236/91310364-b4fa1900-e7a9-11ea-864b-80be538dcd0e.png)


On the **customers side,** the "ready for field" is visible, in the shopfront:
![image](https://user-images.githubusercontent.com/49817236/91310643-0a362a80-e7aa-11ea-9c80-b4deff252d18.png)

On checkout, both "#shipping method delivery instructions" and "OC ready for #outgoing" are visible on the field next to the chosen shipping method:

![image](https://user-images.githubusercontent.com/49817236/91310488-d6f39b80-e7a9-11ea-8e0c-a6006e282958.png)

After checking out, both:
- "#shipping method delivery instructions"
- "OC ready for #outgoing"

appear on the order confirmation Email - see below:

![image](https://user-images.githubusercontent.com/49817236/91314688-bd088780-e7ae-11ea-88c7-8ef2feef5a66.png)

But none of these appears in the PDF versions. Let's see:

PDF invoice - **non-alternative version** - VAT Total included

![image](https://user-images.githubusercontent.com/49817236/91312484-3357ba80-e7ac-11ea-99e3-56acd7dd8e9b.png)

![image](https://user-images.githubusercontent.com/49817236/91313469-55057180-e7ad-11ea-9628-67b34877877f.png)

PDF invoice - **alternative version**  - VAT Total excluded

![image](https://user-images.githubusercontent.com/49817236/91312244-ebd12e80-e7ab-11ea-8c17-7605e0d4b3a8.png)

![image](https://user-images.githubusercontent.com/49817236/91313692-9007a500-e7ad-11ea-8dc8-e486b5d82de6.png)



So, I think the information displayed on the top-right corner of the non-alternative/default version of the invoice (please see green circle above) does not relate to the "ready for" information, but rather to the Order Cycle name (label "OC name #/order_cycles"). If this should not happen, then perhaps we need a separate issue to remove this information.

What do you think @RachL and @tschumilas ?


So, if I understood correctly, the desired outcome of this issue is to include, on both invoices (alternative and default):
- shipping method description, as set on the respective shipping method page
- "ready for" information, as set on the OC Outgoing section

I've changed the title accordingly - please let me know if this is correct :+1: 

Additional question: Should both invoices include "pick-up instructions", as set on the OC Outgoing section, as well? This seems to appear on the checkout page, for the customer.

**Comment by RachL** [comment]
> Should both invoices include "pick-up instructions", as set on the OC Outgoing section, as well? This seems to appear on the checkout page, for the customer. 

I need to check but this rings a bell, I wonder if we haven't got an issue somewhere about this.

> If this should not happen, then perhaps we need a separate issue to remove this information.

yeah that's why I've tagged @lin-d-hop and @kirstenalarsen because to my knowledge the first template (with OC name) is the template used in UK/AU.

the second one without the info, is the one used in France, so all good on my side :trollface: Which template are you using @tschumilas ?

**Comment by Erioldoesdesign** [comment]
This is a great issue so first, just thanks for everyone who provided so much detail! Hopefully I've understood this well enough to comment from a user exp perspective.

Looking back at the question of 'Is this a papercut?'

Sort of 'Yes' if we simply add the shipping method info into an or all PDF invoice templates. Wherever makes the most sense (top right agreed or perhaps at the bottom after the money amounts)

The reason its sort of yes is because of:
1 - Do we make it appear on the default version or persist across all versions because it's super important. Meaning alternate invoice templates.

2 - Do we make it optional/configurable as a choice to display.

Number 1 is 0 design time because the info just goes where it fits with reasonable common sense. Number 2 or anything config/editable etc. I suspect takes longer than 1/2 a day of dev time to do and needs some design time. Not a lot but still enough to make sure users can figure out what's going on so not a papercut.

[This discourse post ](https://community.openfoodnetwork.org/t/add-customizable-information-to-pdf-invoices/2005) 'Add customizable information to PDF invoices' is for sure (even by the title) not a papercut! this discourse post suggests to me a large exploration into what invoices need to do, how, why and how many options over what locations! That is a research and inception piece so I'm taking some comments over to that post 😄

**Comment by tschumilas** [comment]
I lost track of this issue for a bit -  and just got back to it today.  I have started to make a file that lists where info on the 2 different invoice templates and on the order confirmation comes from.   I did this just for my own info (for answering user questions) but now I wonder if it might be something you also take  a look at @filipefurtad0  (and others).    I think we can go to one invoice template -  but we need to figure out what content SHOULD be on it.

These 2 templates were created because we don't have line tax amounts when prices do not include taxes (North American instances in particular).  But does it really matter?  In NA, the amount of tax by line is not going to show on EITHER of these templates anyway. The total tax is accurate no matter which template is used. And the EU/UK/AUS template at least presents the tax rate beside the line item -  which in my view is better to have than nothing.  And frankly,  the invoice template being used in Europe,  UK,  Aus (I think) -  is much  nicer in many other ways - logo,  nicer layout.....  If we could just agree on what content should be there -  I think maybe we can go to one invoice template?

And to your above point @filipefurtad0  -  you are correct,  right now the NA invoice template includes order cycle NAME.  We use this because  'order for' field is NOT on the invoice.   If we pulled 'order for' info,  we don't need 'order cycle name'.  Second,  the NA invoice pulls  Shipping Method info - and we need that,   but the NA invoice does this poorly.  I note that even when the shipping method selected is 'pickup' -  the invoice still says 'shipping' - so that's confusing.  Lets pull shipping method info appropriately for 'delivery'  versus 'pickup'  options.     The final important difference in the 2 invoice templates is the email address of the seller  -  the NA invoice pulls the 'notifications' address,   the EU/UK/AUS template pulls the public email address.   I'm sure we could agree on which it should pull.   

Anyway - here 's a file to start us thinking on this. I included order confirmation content too (I know thats not this particular issue - but it seemed to make sense to put this all on one page -  at least in terms of user support

[https://docs.google.com/spreadsheets/d/1fVSUbJ6KaNY782bNN3GISiJzc1gk_aYz/edit#gid=92166127](url)

**Comment by filipefurtad0** [comment]
This is great @tschumilas , thank you so much for sharing this document. It will surely aid the decision making process. Maybe this should be added to the [post on discourse](https://community.openfoodnetwork.org/t/add-customizable-information-to-pdf-invoices/2005)? 

Indeed the logo is not being displayed for the non-alternative version, regardless of the selection under "Business Details" - thanks for pointing this out. This would need an issue, but perhaps it's best to wait for the outcome of the discussion around the need to have one or two invoice templates.

**Comment by tschumilas** [comment]
He he -  I started a discourse post and then remembered this issue...  -  but I also posted on slack (instance managers)  and I think maybe there is intereste in moving to one invoice template.  So -  I'll do a discourse post around that question so others can join in.  So there is no need to do anything to the non-alternative version right now.  I also just want to double check how the alternative looks for a variety of orders (taxed & non taxed items,  partially paid orders...).  I'm sure its fine - but I really only did a few quick tests yesterday - so just want to be a bit more thorough before I confirm that we can go to one invoice template.
